core/encoding/json/parser.odin

```diff
@@ -263,8 +263,17 @@ parse_object_body :: proc(p: ^Parser, end_token: Token_Kind) -> (obj: Object, er
 			return
 		}
 
-		obj[key] = elem
-		
+		// NOTE(gonz): There are code paths for which this traversal ends up
+		// inserting empty key/values into the object and for those we do not
+		// want to allocate anything
+		if key != "" {
+			reserve_error := reserve(&obj, len(obj) + 1)
+			if reserve_error == mem.Allocator_Error.Out_Of_Memory {
+				return nil, .Out_Of_Memory
+			}
```

Comment:
An `else` would probably be reasonable here but for now it's just this. We weren't checking *anything* before so even this is better than the status quo.

Reply:
Any reason this isn't
```odin
reserve(&obj, len(obj) + 1) or_return
```